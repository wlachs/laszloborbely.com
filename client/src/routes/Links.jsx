import * as React from 'react';
import routes from './index';
import PureLinkHighlight from '../components/PureLinkHighlight';

function Links() {
  return routes.filter((route) => route.display).map((Route) => (
    <PureLinkHighlight key={Route.id} to={Route.path}>{Route.name}</PureLinkHighlight>
  ));
}

export default Links;
