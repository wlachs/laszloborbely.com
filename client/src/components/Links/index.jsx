import * as React from 'react';
import routes from '../../routes/index';
import PureLinkHighlight from '../PureLinkHighlight';

function Links() {
  return routes.filter((route) => route.display).map((Route) => (
    <PureLinkHighlight key={Route.id} to={Route.path}>{Route.name}</PureLinkHighlight>
  ));
}

export default Links;
