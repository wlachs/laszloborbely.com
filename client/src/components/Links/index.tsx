import React from 'react';
import PureLinkHighlight from '../PureLinkHighlight/index.jsx';
import {type RouteProps, routes} from '../../routes';

function Links() {
	const elements = routes
		.sort((a, b) => Number(b.id) - Number(a.id))
		.filter((route: RouteProps) => route.display)
		.map((Route: RouteProps) => (
			<PureLinkHighlight key={Route.id} to={Route.path}>{Route.name}</PureLinkHighlight>
		));

	return (
		<React.Fragment>
			{elements}
		</React.Fragment>
	);
}

export default Links;
