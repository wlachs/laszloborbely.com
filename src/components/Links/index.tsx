import { Fragment, ReactElement } from 'react';

import { type RouteProps, routes } from '../../routes';
import { PureLinkHighlight } from '../PureLinkHighlight';

export function Links(): ReactElement {
	const elements = routes
		.toSorted((a, b) => Number(b.id) - Number(a.id))
		.filter((route: RouteProps) => route.display)
		.map((route: RouteProps) => (
			<PureLinkHighlight key={route.id} to={route.path}>
				{route.name}
			</PureLinkHighlight>
		));

	return <Fragment>{elements}</Fragment>;
}
