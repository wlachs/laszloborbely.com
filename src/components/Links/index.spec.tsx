import {cleanup, render, screen} from '@testing-library/react';
import {MemoryRouter} from 'react-router-dom';
import {afterEach, describe, expect, it, vi} from 'vitest';
import Links from './index.tsx';

vi.mock('../../routes', () => ({
	routes: [
		{
			id: '2',
			display: true,
			name: 'Route #2',
			path: '/2',
			element: <div/>,
		},
		{
			id: '1',
			display: false,
			name: 'Route #1',
			path: '/hidden',
			element: <div/>,
		},
		{
			id: '3',
			display: true,
			name: 'Route #3',
			path: '/3',
			element: <div/>,
		},
		{
			id: '0',
			display: true,
			name: 'Route #0',
			path: '/0',
			element: <div/>,
		},
	],
}));

describe('Links component', () => {
	afterEach(() => {
		cleanup();
	});

	it('should match snapshot', () => {
		const {container} = render(
			<MemoryRouter>
				<Links/>
			</MemoryRouter>,
		);
		expect(container).toMatchSnapshot();
	});

	it('should only show routes with display property', () => {
		const {container} = render(
			<MemoryRouter>
				<Links/>
			</MemoryRouter>,
		);
		expect(screen.queryByText('Route #1')).toBeNull();
		expect(container.children.length).toEqual(3);
	});

	it('should sort routes by ID', async () => {
		const {container} = render(
			<MemoryRouter>
				<Links/>
			</MemoryRouter>,
		);
		const res = screen.getAllByText(/Route #[0-9]/);
		const labels = res.map(r => r.innerHTML);
		expect(labels).toEqual(['Route #3', 'Route #2', 'Route #0']);
	});
});
