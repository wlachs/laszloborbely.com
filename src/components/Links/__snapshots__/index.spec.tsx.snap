// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Links component > should match snapshot 1`] = `
<div>
  <a
    class="link"
    href="/3"
  >
    Route #3
  </a>
  <a
    class="link"
    href="/2"
  >
    Route #2
  </a>
  <a
    class="link"
    href="/0"
  >
    Route #0
  </a>
</div>
`;
